### PR TITLE
Some swizzle changes split out from #1900

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Split out from #1900

### CODE REVIEW

**Review by csarofeen — APPROVED** [review]
LGTM

**Comment by csarofeen** [comment]
@zasdfgbnm was marked as draft hopefully it was ready to merge, I don't see any concerns with it.

**Comment by naoyam** [comment]
Why is this split out of #1900? Looks like I had a couple of questions on #1900, and some of them may be included here.

**Comment by csarofeen** [comment]
My fault, didn't know there were pending comments. Which ones would you have considered blocker?

File: torch/csrc/jit/codegen/cuda/lower_validation.cpp [hunk 1]
```diff
@@ -288,6 +288,13 @@ class VectorizeValidator : public OptInDispatch {
     domains_.insert(m->inner());
   }
 
+  void handle(Swizzle2D* swizzle) final {
+    if (swizzle->outX() == vectorized_id_ || swizzle->inX() == vectorized_id_) {
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/lower_validation.cpp:292]: I think this is probably the worst offender @naoyam spotted that I missed.

**Comment by zasdfgbnm** [comment]
We can fix those in a follow-up remaining comments. I was digging into the codebase to see how swizzles are lowered, that's why I marked it as WIP.

**Comment by naoyam** [comment]
I generally wasn't comfortable with the #1900 yet, mostly on the changes in matmul.cpp. Looks like this PR didn't take that part, but I wasn't sure why only some part was taken here.

Another thing I wasn't clear was the skipping logic in BestEffortReplay. https://github.com/csarofeen/pytorch/pull/1900#discussion_r976024111

**Comment by zasdfgbnm** [comment]
> mostly on the changes in matmul.cpp.

I wasn't comfortable about these changes in `matmul.cpp` either; that's the only reason why I am splitting the swizzle part out as a separate PR.

I want to consider the improvements on swizzle and scheduling of matmul separate topic, and handle them separately. I might do similar things a lot in the future: I will cherry-pick whatever I am feeling more comfortable with to make them separate PR, so we can merge them early without blocking on other parts.

My original plan for this PR was to read through the entire indexing and other passes to get a complete understanding on how swizzles are lowered. But I think it is OK that this PR is merged as is, because I don't think this will break anything. We can continue the review here or at #1900. And I will submit new changes in a separate PR.

**Pull request merged**